### PR TITLE
Main feature gamedata

### PR DESCRIPTION
**Pull request by Dexusan**
- Implement bepinex root gamedata path with legacy support
- changed project settings
- cleaned up some classes

### CODE REVIEW

**Comment by Dexusan** [comment]
Please note this is a half baked implementation, there are plans to reduce the codebase in the future

File: MTFO/Managers/ConfigManager.cs [hunk 1]
```diff
@@ -43,12 +43,17 @@ static ConfigManager()
...
+            {
+                GameDataPath = ResolveGameDataPath(Paths.PluginPath);
+                Log.Warn("Plugin paths for gamedata are under legacy support and will be removed in the future. Considering migrating to the '\\BepInEx\\GameData' folder.");
+            }
 
             if (HasGameDataPath)
             {
                 CustomPath = Path.Combine(GameDataPath, CUSTOM_FOLDER);
                 HasCustomContent = Directory.Exists(CustomPath);
+                Log.Warn("Consider migrating custom json data to the '\\BepInEx\\GameData folder instead");
```
Comment (Kasuromi) [inline MTFO/Managers/ConfigManager.cs:56]: This warning will be thrown for either path usage. Maybe add a check to make sure that CustomPath is targeting legacy.

**Review by Kasuromi — CHANGES_REQUESTED** [review]

File: MTFO/Managers/ConfigManager.cs (hunk 1 above)
Comment (Dexusan) [reply MTFO/Managers/ConfigManager.cs:56]: The intent is to nudge plugin developers away from using the MTFO custom path, since the bepinex gamedata path

Comment (Dexusan) [reply MTFO/Managers/ConfigManager.cs]: Removed warning instead

**Review by Kasuromi — APPROVED** [review]

**Pull request merged**